### PR TITLE
Bug Fixes | aave

### PR DESCRIPTION
**Pull request by pradyuman-verma**
- Fixed swapBorrowMode bug in aave v2, v3
- [x] Review @thrilok209 
- [x] deploy
- [x] enable
- [x] Add to SDK https://github.com/Instadapp/dsa-connect/pull/192

### CODE REVIEW

**Review by thrilok209 — DISMISSED** [review]

**Comment by iri031** [comment]
## Tests [ Polygon ]

Screenshot from 2022-06-18 23-07-45

## Tests [ Arbitrum ]

Screenshot from 2022-06-18 22-45-49

## Tests [ Avalanche ]

Screenshot from 2022-06-18 22-30-15

## Tests [ Optimism ]

Screenshot from 2022-06-18 22-54-07

## Tests [ Fantom ]

Screenshot from 2022-06-18 23-13-03

**Comment by pradyuman-verma** [comment]
@Richa-iitr Let's deploy.

**Comment by iri031** [comment]
## Deployed connector for Polygon 

**Name** `AAVE-V3-A`
**Addr** [0x68765CBa5B4269d22036B33E5da65a68F719dA7A](https://polygonscan.com/address/0x68765CBa5B4269d22036B33E5da65a68F719dA7A#code)
**tx** [here](https://polygonscan.com/tx/0x0890ae842e0fc3d3664ed6704970c99f787a4f40a2cc6cfc428eb9bd6ac8f360#eventlog)

## Deployed connector for Arbitrum

**Name** `AAVE-V3-A`
**Addr** [0x09cadCBe6Fe623F01B7bC37635b97ABF385d46Dd](https://arbiscan.io/address/0x09cadCBe6Fe623F01B7bC37635b97ABF385d46Dd#code)
**tx** [here](https://arbiscan.io/tx/0x5e7e25a475db15254874bc0116fd789084c04f8984c5f47b610a3fd3bfcb0d2f#eventlog)

## Deployed connector for Avalanche

**Name** `AAVE-V3-A`
**Addr** [0xE3D068954336fEF7A6A21056431df2E1044f1A5C](https://snowtrace.io/address/0xE3D068954336fEF7A6A21056431df2E1044f1A5C#code)
**tx** [here](https://snowtrace.io/tx/0xc0ea68ab1945c3ee4916ebbdf6c5493b70cb2db703ed235cb46af5b55fc6d043#eventlog)


## Deployed connector for Optimism

**Name** `AAVE-V3-A`
**Addr** [0x89566CD6302EC9730970BB4E9374d9329b756E2C](https://optimistic.etherscan.io/address/0x89566CD6302EC9730970BB4E9374d9329b756E2C#code)
**tx** [here](https://optimistic.etherscan.io/tx/0x992282e398204d525f2acb745f969b61152ef32ccda71f8edfa46f0007f331de#eventlog)

## Deployed connector for Fantom

**Name** `AAVE-V3-A`
**Addr** [0x44aC5eD26d8B7BBcF3Bcb9Ac486191cBb294B140](
https://ftmscan.com/address/0x44aC5eD26d8B7BBcF3Bcb9Ac486191cBb294B140#code)
**tx** [here](https://ftmscan.com/tx/0xf037c82e063c3fb78030f2c236f0f7217cc724f2080aa66e8576f7b0ce113128#eventlog)

File: contracts/mainnet/connectors/aave/v2/main.sol [hunk 1]
```diff
@@ -214,56 +240,69 @@ abstract contract AaveResolver is Events, Helpers {
...
-        _eventName = "LogSwapRateMode(address,uint256)";
-        _eventParam = abi.encode(token, rateMode);
-    }
+	/**
+	 * @dev Swap borrow rate mode
+	 * @notice Swaps user borrow rate mode between variable and stable
+	 * @param token The address of the token to swap borrow rate.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param currentRateMode Current Rate mode. (Stable = 1, Variable = 2)
+	 */
+	function swapBorrowRateMode(address token, uint256 currentRateMode)
```
Comment (thrilok209) [inline contracts/mainnet/connectors/aave/v2/main.sol]:
```suggestion
	function swapBorrowRateMode(address token, uint256 rateMode)
```

File: contracts/mainnet/connectors/aave/v2/main.sol [hunk 2]
```diff
@@ -214,56 +240,69 @@ abstract contract AaveResolver is Events, Helpers {
...
+
+		bool isEth = token == ethAddr;
+		address _token = isEth ? wethAddr : token;
+
+		if (getPaybackBalance(_token, currentRateMode) > 0) {
+			aave.swapBorrowRateMode(_token, currentRateMode);
+		}
+
+		_eventName = "LogSwapRateMode(address,uint256)";
+		_eventParam = abi.encode(token, currentRateMode);
```
Comment (thrilok209) [inline contracts/mainnet/connectors/aave/v2/main.sol]:
```suggestion
		if (getPaybackBalance(_token, rateMode) > 0) {
			aave.swapBorrowRateMode(_token, rateMode);
		}

		_eventName = "LogSwapRateMode(address,uint256)";
		_eventParam = abi.encode(token, rateMode);
```

File: contracts/mainnet/connectors/aave/v2/main.sol (hunk 1 above)
Comment (iri031) [reply contracts/mainnet/connectors/aave/v2/main.sol]: Updated

File: contracts/mainnet/connectors/aave/v2/main.sol (hunk 2 above)
Comment (iri031) [reply contracts/mainnet/connectors/aave/v2/main.sol]: Updated

**Comment by pradyuman-verma** [comment]
### Deployed on Polygon

Name: `AAVE-V2-A`
Addr: [0x14272cF069a5cE1B97c9999B8c368Cf3704aCD0B](https://polygonscan.com/address/0x14272cF069a5cE1B97c9999B8c368Cf3704aCD0B)
Tx: [here](https://polygonscan.com/tx/0xb81d917bf762eb9cacc6b761b4c6206ecee5fd04b7c4f3a2656ac913c871e41b#eventlog)

### Deployed on Avalanche

Name: `AAVE-V2-A`
Addr: [0xa4361Ff34949bEf0605a10ad7B50fE1e22C92E3B](https://snowtrace.io/address/0xa4361Ff34949bEf0605a10ad7B50fE1e22C92E3B#code)
Tx: [here](https://snowtrace.io/tx/0xc0ea68ab1945c3ee4916ebbdf6c5493b70cb2db703ed235cb46af5b55fc6d043#eventlog)


### Deployed on mainnet
Name: `AAVE-V2-A`
Addr: [0xdb8e3f36c6f4368cb2c023f41225428e366b0b2a](https://etherscan.io/address/0xdb8e3f36c6f4368cb2c023f41225428e366b0b2a)
tx: [here](https://etherscan.io/tx/0x4724f754bdaaaa02f274ba7b9dfd831c8477a5e710291a5f736bd71ea6ec37f0#eventlog)

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**